### ISSUE TITLE
why you use two optimizers for E and G?

### ISSUE THREAD

**Issue by cwellszhang** [HotaekHan/Synthetically_Supervised_Text_Recognition#1]
in original paper, they are claimed to be jointly optimized.  
BTW,  since there is no Readme, I wonder could this code obtain the same result claimed in the paper?.

**Comment by HotaekHan** [comment]
@cwellszhang Hello :) thanks for comment!
as my knowledge, my code jointly update encoder and generator. because loss is accumulated from both sequence loss and reconstruction loss.
I’m trying to reproduce the results from paper. but not easy :( because synth dataset is so big and hard to find lambda for each loss.

**Comment by cwellszhang** [comment]
@HotaekHan  Thanks for your reply.  I understood it is  a very tough work.
I found that the L2 loss gradient will backprop twice from the real image and synth image, will that be unstable?   What are the differences if I detach the synth image or the real image (using .detach())?

It  seems that the performance will be close to the SOTA with only L2 loss,  have you found the best lambda for that loss ?   If not yet,  from my suggestion,  it will be much easier and faster than other losses  as it needs no additional structures.

**Comment by HotaekHan** [comment]
@cwellszhang I can't find where you mentioned. could you tell me which line?
L2 loss is work for feature matching.

hmm..I can't get what you mean :p  
only L2 loss is enough for performance?? what is your suggestion??

**Comment by cwellszhang** [comment]
I mean, the code is complicated. If I wanna try whether it works, I will try L2 loss first.
something that confused me is that:  
in line235  : feat_match_loss = feat_criterion(encoded_synth, encoded_real)
what will happen if i use encoded_real.detach() or encoded_synth.detach() ?

**Comment by HotaekHan** [comment]
@cwellszhang yes, code is complicated :p because several losses are jointly updated.
if you detach the encoded image, encoder E is not updated by L2 loss.
if you want to run this code, you should download the dataset first. And make the dataset using data/create_listdata_VGG90.py.
And you can configure the hyperparameter using config file(yaml).
check out configs/synthVGG.yaml.
you can change the lambda values for each loss.

**Comment by HotaekHan** [comment]
I will update README as soon as possible

**Comment by cwellszhang** [comment]
Well done, thanks a lot :)
